src/libcore/slice/mod.rs

```diff
@@ -1364,6 +1364,20 @@ impl<T> [T] {
     /// let r = s.binary_search(&1);
     /// assert!(match r { Ok(1..=4) => true, _ => false, });
     /// ```
+    ///
+    /// If you want to insert an item to a sorted vector, while maintaining
+    /// sort order:
+    ///
+    /// ```
+    /// let mut s = vec![0, 1, 1, 1, 1, 2, 3, 5, 8, 13, 21, 34, 55];
+    /// let num = 42;
+    /// let idx = match s.binary_search(&num) {
+    ///     Ok(idx) => idx,
+    ///     Err(idx) => idx,
```

Comment:
The arms could be simplified by using `Ok(idx) | Err(idx) => idx`

Reply:
I personally find this clearer than that but r=me with this changed as well.

Reply:
This should be further simplified to `let Ok(idx) | Err(idx) = s.binary_search(&num);` once we have or-patterns, https://github.com/rust-lang/rust/issues/54883.

Reply:
I use `unwrap_or_else`. Do you prefer it instead?